add watch.excludes to exclude paths

PXshadow opened the pull request:
Excludes a path using StringTools startsWith. It's useful for when you have a project that generates Haxe files such as a compiler.

PXshadow committed:
add watch.excludes to exclude paths

benmerckx commented:
This is useful, but I wonder how the implementation can deal with:
- spaces in file/dir names
- relative paths
- mixed directory separators (/ and \\)
- case sensitivity in windows, cases like C:\... vs c:\...

benmerckx commented:
I've addressed the points above in 70ec56bbe82150871214d99a3884329453bb3d01
- Paths are normalized and can be relative
- Excluding multiple paths can be done by using the define repeatedly: `-D watch.exclude=a -D watch.exclude=b`

PXshadow commented:
> I've addressed the points above in [70ec56b](https://github.com/benmerckx/watch/commit/70ec56bbe82150871214d99a3884329453bb3d01)
> 
> * Paths are normalized and can be relative
> * Excluding multiple paths can be done by using the define repeatedly: `-D watch.exclude=a -D watch.exclude=b`

Thanks so much, sorry didn't managage to see the messages. Looks like it should work great 👍
